akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -1,867 +1,603 @@
+
+      val actor = spawn(behavior)
+      actor ! Ping
+      probe.expectMessage(Pong)
+      actor ! Miss
+      probe.expectMessage(Missed)
+      actor ! Ping
+      probe.expectMessage(Pong)
     }
-  }
 
-  def muteExpectedException[T <: Exception: ClassTag](
-    message:     String = null,
-    source:      String = null,
-    start:       String = "",
-    pattern:     String = null,
-    occurrences: Int    = Int.MaxValue)(implicit system: ActorSystem[GuardianCommand]): EventFilter = {
-    val filter = EventFilter(message, source, start, pattern, occurrences)
-    import scaladsl.adapter._
-    system.toUntyped.eventStream.publish(Mute(filter))
-    filter
-  }
-
-  // for ScalaTest === compare of Class objects
-  implicit def classEqualityConstraint[A, B]: CanEqual[Class[A], Class[B]] =
-    new CanEqual[Class[A], Class[B]] {
-      def areEqual(a: Class[A], b: Class[B]) = a == b
-    }
-
-  implicit def setEqualityConstraint[A, T <: Set[_ <: A]]: CanEqual[Set[A], T] =
-    new CanEqual[Set[A], T] {
-      def areEqual(a: Set[A], b: T) = a == b
-    }
+    "correctly wire the lifecycle hook" in {
+      val probe = TestProbe[Event]()
 
-  /**
-   * The name for the set of tests to be instantiated, used for keeping the test case actors’ names unique.
-   */
-  def suite: String
-
-  /**
-   * The behavior against which to run all the tests.
-   */
-  def behavior(ctx: scaladsl.ActorContext[Event], ignorePostStop: Boolean): Behavior[Command]
-
-  private def mySuite: String = suite + "Adapted"
-
-  def setup(name: String, wrapper: Option[Behavior[Command] ⇒ Behavior[Command]] = None, ignorePostStop: Boolean = true)(
-    proc: (scaladsl.ActorContext[Event], StepWise.Steps[Event, ActorRef[Command]]) ⇒ StepWise.Steps[Event, _]): Future[Status] =
-    runTest(s"$mySuite-$name")(StepWise[Event] { (ctx, startWith) ⇒
-      val b = behavior(ctx, ignorePostStop)
-      val props = wrapper.map(_(b)).getOrElse(b)
-      val steps = startWith.withKeepTraces(true)(ctx.spawn(props, "subject"))
-
-      proc(ctx, steps)
-    })
-
-  private implicit class MkC(val startWith: StepWise.Steps[Event, ActorRef[Command]]) {
-    /**
-     * Ask the subject to create a child actor, setting its behavior to “inert” if requested.
-     * The latter is very useful in order to avoid disturbances with GotSignal(PostStop) in
-     * test procedures that stop this child.
-     */
-    def mkChild(
-      name:    Option[String],
-      monitor: ActorRef[Event],
-      self:    ActorRef[Event],
-      inert:   Boolean         = false): StepWise.Steps[Event, (ActorRef[Command], ActorRef[Command])] = {
-      val s =
-        startWith.keep { subj ⇒
-          subj ! MkChild(name, monitor, self)
-        }.expectMessage(expectTimeout) { (msg, subj) ⇒
-          val Created(child) = msg
-          (subj, child)
-        }
+      val internal = (Behaviors.immutablePartial[Command] {
+        case (_, Fail) ⇒
+          throw new RuntimeException("Boom")
```

Comment:
We have a special `akka.actor.typed.TestException` that doesn't keep a stack trace, please use that (even if the original test did not)

Also, the original test muted the expected exception errors from the log, but let's do a separate pass to fix that, since this is already hard enough to review and that more of a mild annoyance with expected errors in the logs.